addons/strimzi/duck/v1beta2/duck_types.go

```diff
@@ -40,6 +40,12 @@ const (
 type KafkaTopic struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              KafkaTopicSpec `json:"spec,omitempty"`
+}
+
+// KafkaTopicSpec is the duck of a KafkaTopic spec.
+type KafkaTopicSpec struct {
```

Comment:
I thought the same but I noticed the gen crd was tagged with `noStatus` (tbh, not sure why), reason why I preferred to be conservative. I can definitely move to the status if somebody confirm it would be the best thing to do.

Reply:
@lburgazzoli I would say `topicName` as this is the name used by the kafka broker to create the topic. kafkaTopic name is only the name of the CR that appears under the api.

Hope that helps ?

Reply:
> I thought the same but I noticed the gen crd was tagged with `noStatus` (tbh, not sure why), reason why I preferred to be conservative. I can definitely move to the status if somebody confirm it would be the best thing to do.

it seems this is for our own processing logic, maybe it is for the RBACs generation. In the CRD definition I see that the staus is defined https://github.com/strimzi/strimzi-kafka-operator/blob/main/install/cluster-operator/043-Crd-kafkatopic.yaml#L77-L110.

Maybe we should ask the Strimzi folks about what is the expected behavior.

Reply:
Changed to `.status.topicName`. It seems to be the right thing to do, thanks.